File: ch09/rdd.scala
import org.apache.spark.sql.SparkSession

val spark = SparkSession.builder.appName("FunctionalExample").getOrCreate()
val sc = spark.sparkContext

// Data: A sequence of integers
val data = Seq(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)

// Parallelize data into an RDD
val rdd = sc.parallelize(data)

// FUNCTIONAL PIPELINE:
// Uses '=>' (Arrow Syntax) as described in Source [1]
val result = rdd.filter(x => x % 2 == 0)  // Keep evens
                .map(x => x * x)          // Square them
                .reduce((a, b) => a + b)  // Sum them

// ALTERNATIVE (More "Functional" Placeholder Syntax):
// val result = rdd.filter(_ % 2 == 0).map(x => x * x).reduce(_ + _)

println(s"Result: $result")

File: ch09/rdd.py
from pyspark.sql import SparkSession

spark = SparkSession.builder.appName("FunctionalExample").getOrCreate()
sc = spark.sparkContext

# Data: A list of integers
data = [2, 4, 6, 8, 10]

# Parallelize data into an RDD
rdd = sc.parallelize(data)

# FUNCTIONAL PIPELINE:
# 1. Filter: Keep only even numbers (x % 2 == 0)
# 2. Map: Square the remaining numbers (x * x)
# 3. Reduce: Sum the results (a + b)
result = rdd.filter(lambda x: x % 2 == 0) \
            .map(lambda x: x * x) \
            .reduce(lambda a, b: a + b)

print(f"Result: {result}")
# Output will be sum of (2^2, 4^2, 6^2, 8^2, 10^2) -> 220